keras/optimizers.py

```diff
@@ -69,6 +70,9 @@ def get_updates(self, params, constraints, loss):
 
     def get_gradients(self, loss, params):
         grads = K.gradients(loss, params)
+        if hasattr(self, 'normalize_gradients') and self.normalize_gradients:
+            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads])) + K.epsilon()
```

Comment:
In that case, I would prefer to have gradient_modifier_fn which can allow all sorts of things including clipping etc. There might be use cases where negative gradients are dropped etc. I think it's a nice option for research purposes.